### PR TITLE
imports: LOC

### PR DESCRIPTION
**Pull request by rerowep**
* Corrects transformation errors for bad searches for LOC.

### CODE REVIEW

File: rero_ils/modules/imports/api.py [hunk 1]
```diff
@@ -375,11 +378,13 @@ def _split_stream(stream):
...
                             self.data.append(json_data)
                             self.results['hits']['hits'].append(data)
-                    self.results['hits']['remote_total'] = int(etree.parse(
-                            BytesIO(response.content))
-                            .find('{*}numberOfRecords').text)
+                    self.results['hits']['remote_total'] = 0
+                    with contextlib.suppress(Exception):
+                        self.results['hits']['remote_total'] = int(etree.parse(
+                                BytesIO(response.content))
+                                .find('{*}numberOfRecords').text)
```
Comment (zannkukai) [inline rero_ils/modules/imports/api.py:385]: I think it's not the pythonic way to use. `contextlib.suppress(Exception)` should be used only as last option.
In this case, a simple `try...except` seems better :
```python
try:
    self.results['hits']['remote_total'] = int(etree.parse(
          BytesIO(response.content))
          .find('{*}numberOfRecords').text)
except Exception:
   self.results['hits']['remote_total'] = 0
```

**Review by zannkukai — CHANGES_REQUESTED** [review]

**Review by zannkukai — APPROVED** [review]

**Review by Garfield-fr — APPROVED** [review]

**Review by jma — CHANGES_REQUESTED** [review]
Can you provide some tests? I would be great to have a more explicit commit message. Thanks.

File: rero_ils/modules/imports/api.py [hunk 2]
```diff
@@ -484,7 +487,10 @@ def get_id(self, json_data):
         :param id: json document
         :return: id of the record
         """
-        return json_data.get('010__').get('a').strip()
+        id = None
+        if json_data.get('010__'):
+            id = json_data.get('010__').get('a').strip()
+        return id
```
Comment (lauren-d) [inline rero_ils/modules/imports/api.py:493]: 1. In python, the default return value for a function is `None`
(see: https://realpython.com/python-return-statement/#:~:text=If%20you%20don%E2%80%99t%20supply%20an%20explicit%20return%20statement%20with%20an%20explicit%20return%20value%2C%20then%20Python%20will%20supply%20an%20implicit%20return%20statement%20using%20None%20as%20a%20return%20value)
so, you can just do :
```python
if json_data.get('010__'):
  return json_data.get('010__').get('a').strip()
```
2. Are you sure that tag `010` always have a subfield `a` ?

**Review by lauren-d — CHANGES_REQUESTED** [review]

File: rero_ils/modules/imports/api.py (hunk 2 above)
Comment (rerowep) [reply rero_ils/modules/imports/api.py:493]: I had already forgotten about it, thank you

**Review by lauren-d — APPROVED** [review]

**Review by jma — APPROVED** [review]

**Pull request merged**